docs/data-product-studio/data-quality/failed-events/exploring-failed-events/warehouse-lake/index.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Exploring failed events in the warehouse or data lake"
+description: "Load common types of failed events to a separate table in your warehouse or lake to analyze them easily."
+sidebar_label: "In warehouse or lake"
+sidebar_position: 1
+---
+
+:::note Compatibility
+
+This feature is available since Enrich 5.0.0 and works with Snowflake and Lake loaders.
+
+:::
+
+## Introduction
+
+[Failed events](/docs/fundamentals/failed-events/index.md) are events the pipeline had some problem processing (for example, events that did not pass validation).
+
+For the common failures (validation and enrichment), you can configure continuous loading of any offending events into _a separate table_ in your warehouse or lake. This way, you can easily inspect them and decide how they might be patched up (e.g. with SQL) and merged with the rest of your data.
+
+:::note
+
+This feature is not retroactive, i.e. only failed events that occur _after it’s enabled_ will be loaded into your desired destination.
+
+:::
+
+## Format
+
+The format of the failed events loaded into your warehouse or lake is [the same as for your atomic events](/docs/fundamentals/canonical-event/index.md). All the standard fields are present (unless themselves invalid — see below). This allows you to query and aggregate this data easily, e.g. if you want to see the number of failures per `app_id`.
+
+There are two differences compared to regular events.
+
+**Invalid data is removed from the event.** This principle applies to all columns:
+* Any invalid standard column (e.g. `geo_country`) will be set to `null`.
```

Comment:
so there would be no more `geo_country` column? or the data in that column is null?

Reply:
It will not delete columns from the warehouse lol! just set the value of the column to `null` for that row. Do you reckon the text is not clear as stated?

Reply:
How about now?